storm-core/src/jvm/backtype/storm/validation/ConfigValidation.java

```diff
@@ -489,17 +489,63 @@ public  void validateField(String name, Object o) {
     public static class PacemakerAuthTypeValidator extends Validator {
         @Override
         public void validateField(String name, Object o) {
-            if(o == null) {
-                throw new IllegalArgumentException( "Field " + name + " must be set.");
+            if (o == null) {
+                throw new IllegalArgumentException("Field " + name + " must be set.");
+            }
+
+            if (o instanceof String &&
+                    (((String) o).equals("NONE") ||
+                            ((String) o).equals("DIGEST") ||
+                            ((String) o).equals("KERBEROS"))) {
+                return;
+            }
+            throw new IllegalArgumentException("Field " + name + " must be one of \"NONE\", \"DIGEST\", or \"KERBEROS\"");
+        }
+    }
```

Comment:
OK, it picked up the space after the `if`: `if(` -> `if (`. I am fine with the changes as they are.